.github/workflows/gradle.yml

```diff
@@ -45,3 +45,6 @@ jobs:
           $Env:Path += ';.'
           gradlew.bat build
           gradlew.bat runFunctionalTest
+      - name: Use java.util.Objects not Guava
+        run: |
+          ! grep -q -r --include \*.java com.google.common.base.Objects
```

Comment:
1. Move this check before `Setup java` so that it fails earlier.
2. The check should be like below
```yaml
- name: Check limited Guava usage
  run: |
    if grep -r com.google.common.base.Objects .; then echo "Error: use java.util.Objects instead of com.google.common.base.Objects" && false; fi
```

Reply:
I've done something like this on my first try and then I thought I will make it "better" :-)